services/galley/src/Galley/API/Create.hs

```diff
@@ -130,10 +134,16 @@ createOne2OneConversation (zusr ::: zcon ::: req) = do
   maybe (create x y n $ newConvTeam j) (conversationResponse status200 zusr) c
   where
     checkBindingTeamPermissions x y tid = do
-      mems <- bindingTeamMembers tid
-      void $ permissionCheck zusr CreateConversation mems
-      unless (all (flip isTeamMember mems) [x, y]) $
-        throwM noBindingTeamMembers
+      zusrMembership <- Data.teamMember tid zusr
+      permissionCheckSimple CreateConversation zusrMembership
+      Data.teamBinding tid >>= \case
+        Just Binding -> do
+          xMembership <- Data.teamMember tid x
+          yMembership <- Data.teamMember tid y
+          unless (all isJust [xMembership, yMembership]) $
+            throwM noBindingTeamMembers
```

Comment:
Here, changing to check the first membership before looking up the second one is quite straight-forward.

What's wrong with me today?  I'm obsessed with performance.  :)

Reply:
Done.